pkg/engine/armvariables.go

```diff
@@ -380,6 +380,14 @@ func getK8sMasterVars(cs *api.ContainerService) (map[string]interface{}, error)
 
 			}
 		} else {
+			if cs.Properties.OrchestratorProfile.KubernetesConfig.LoadBalancerSku == api.StandardLoadBalancerSku && hasAgentPool {
```

Comment:
should some of these be omitted for private clusters?

Reply:
The outer if/else checks for IsPrivateCluster already https://github.com/Azure/aks-engine/blob/2afb04123eca2fc28507683b205c31949be9dd5e/pkg/engine/armvariables.go#L356